avocado/utils/linux_modules.py

```diff
@@ -96,7 +101,23 @@ def loaded_module_info(module_name):
     :rtype: dict
     """
     l_raw = process.system_output('/sbin/lsmod')
-    return parse_lsmod_for_module(l_raw, module_name)
+    modinfo_dic = parse_lsmod_for_module(l_raw, module_name)
+    output = process.system_output("/sbin/modinfo %s" % module_name)
+    if output:
+        param_list = []
+        for line in output.splitlines():
+            key = line.split()[0]
+            if key == 'filename:':
+                modinfo_dic['filename'] = line.split()[-1]
```

Comment:
Line 109 splits the `line`. One of the lines 111, 113, 115&116 or 118 can split the `line` **again**. Since all modules have, at least, `filename` and `depends`, I'm suggesting to avoid here, at least, 2 extra `line` splits (3 when `depends` has something in it).